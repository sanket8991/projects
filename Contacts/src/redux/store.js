import { configureStore } from "@reduxjs/toolkit";
import contactsReducer from './contactSlice';

const store = configureStore({
    reducer:{contacts: contactsReducer}
});

export default store;
