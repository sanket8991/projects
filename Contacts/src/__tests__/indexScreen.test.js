import React from 'react';
import IndexScreen from "../screens/IndexScreen";
import {render, fireEvent} from '@testing-library/react-native';
import store from '../redux/store';
import {Provider} from 'react-redux';

test('should render the screen properly', () => {
    const {toJSON} = render(<Provider store = {store}><IndexScreen/></Provider>)
    expect(toJSON()).toMatchSnapshot();
})
test('testing the delete icon', () => {
    const {getByTestId} = render(<Provider store = {store}><IndexScreen/></Provider>);
    const onPressMock = jest.fn();

    const foundIcon = getByTestId("deleteIcon");
    expect(foundIcon).toBeTruthy();
    fireEvent.press(foundIcon);
    
    
});