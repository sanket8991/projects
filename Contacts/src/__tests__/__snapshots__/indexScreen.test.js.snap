// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the screen properly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#D7D7D7",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 50,
        "maginBottom": 10,
        "marginHorizontal": 10,
        "marginTop": 10,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search name..."
      testID="searchInput"
      value=""
    />
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "contact": "123456789",
          "email": "john@email.com",
          "id": "m6mCM9WiOxTLKloREk1y_",
          "imagePath": "",
          "name": "john doe",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    keyExtrator={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "grey",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "paddingVertical": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
               
              john doe
            </Text>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="deleteIcon"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
